File: Rust/Iniciante/1049.rs
use std::io;
use std::error::Error;

fn main() -> Result<(), Box<dyn Error>> {
	let stdin = io::stdin();

	let mut c1 = String::new();
	let mut c2 = String::new();
	let mut c3 = String::new();

	stdin.read_line(&mut c1)?;
	stdin.read_line(&mut c2)?;
	stdin.read_line(&mut c3)?;

	let rc1 = c1.trim().as_ref();
	let rc2 = c2.trim().as_ref();
	let rc3 = c3.trim().as_ref();

	let classificacao = match rc1 {
		"vertebrado" => match rc2 {
			"ave" => match rc3 {
				"carnivoro" => "aguia",
				"onivoro"   => "pomba",
				_           => "",
			}
			"mamifero" => match rc3 {
				"onivoro"   => "homem",
				"herbivoro" => "vaca",
				_           => "",
			}
			_ => "",
		}
		"invertebrado" => match rc2 {
			"inseto" => match rc3 {
				"hematofago" => "pulga",
				"herbivoro"  => "lagarta",
				_            => "",
			}
			"anelideo" => match rc3 {
				"hematofago" => "sanguessuga",
				"onivoro"    => "minhoca",
				_            => "",
			}
			_ => "",
		}
		_ => "",
	};

	println!("{}", classificacao);

	Ok(())
}

File: Python/Iniciante/1049.py
# Problema 1049 - Beecrowd - Iniciante - Nível 3

# Vamos receber primeiro as três strings, antes de classificar
filo = input()
classe = input()
alimentacao = input()

# Agora, vamos classificar utilizando IF's aninhados:
if filo == 'vertebrado':
    if classe == 'ave':
        if alimentacao == 'carnivoro':
            print('aguia')
        elif alimentacao == 'onivoro':
            print('pomba')
    elif classe == 'mamifero':
        if alimentacao == 'onivoro':
            print('homem')
        elif alimentacao == 'herbivoro':
            print('vaca')

elif filo == 'invertebrado':
    if classe == 'inseto':
        if alimentacao == 'hematofago':
            print('pulga')
        elif alimentacao == 'herbivoro':
            print('lagarta')
    elif classe == 'anelideo':
        if alimentacao == 'hematofago':
            print('sanguessuga')
        elif alimentacao == 'onivoro':
            print('minhoca')